### PR TITLE
[Competitive programming]: K- Concatenation Problem codechef

### LINKED ISSUE #5052: [Competitive programming]:  K- Concatenation Problem codechef
### Description

You are given an array A with size N and an integer K. Another array B with size N*K is created with concatenation of the array A, K times.
Find the maximum subarray sum of array B.
I will be providing documentation with explanation of code using C++

### Domain

Competitive Programming

### Type of Contribution

Documentation

### Code of Conduct

- [X] I follow [Contributing Guidelines](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CONTRIBUTING.md) & [Code of conduct](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CODE_OF_CONDUCT.md) of this project.

### PR DESCRIPTION
**Pull request by vishvarana**
<hr>

## Description 📜

Added solution to K-concatenation problem from codechef using C++ language

Fixes #5052

<hr>

## Type of change 📝

- [ ] Au#dio (Should be in mp3 format Includes speech clarity, Concise ,Low distortion)
- [ ] Vi#deo (Animations, screen-recordings, presentations and regular explanatory films are all possibilties etc)
- [√] Documentation (Content Creation in the form of codes or tutorials)
- [ ] Other (If you choose other, Please mention changes below) 

<hr>

## Domain of Contribution 📊

- [√] Competitive #Programming
<hr>

## Checklist ✅

- [√] I follow [Contributing Guidelines](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CONTRIBUTING.md) & [Code of conduct](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CODE_OF_CONDUCT.md) of this project.
- [√] I have performed a self-review of my own code or work.
- [√] I have commented my code, particularly in hard-to-understand areas.
- [√] My changes generates no new warnings.
- [√] I'm GWOC'21 contributor

<hr>

### CODE REVIEW

**Review by ravireddy07 — CHANGES_REQUESTED** [review]
before going to the code, Please explain the approach in words that you used to solve this problem.
And add Complexity details along with few testcase examples.

File: Competitive_Programming/C++/Array/K_concatenation_prob.md [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+3 3
+
+1 3 -2 
+
+**Output 2**
+
+6
+
+
+**Complexities**
```
Comment (ravireddy07) [inline Competitive_Programming/C++/Array/K_concatenation_prob.md:128]: Use header(#) here instead of Bold

Comment (ravireddy07) [inline Competitive_Programming/C++/Array/K_concatenation_prob.md:128]:
```suggestion
## Complexity
```

File: Competitive_Programming/C++/Array/K_concatenation_prob.md [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+
+1 -2 1
+
+**Output 1**
+
+9
+
+2
+
+**Explanation**
```
Comment (ravireddy07) [inline Competitive_Programming/C++/Array/K_concatenation_prob.md:111]:
```suggestion
## Explanation
```

File: Competitive_Programming/C++/Array/K_concatenation_prob.md [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+
+3 2
+
+1 -2 1
+
+**Output 1**
+
+9
+
+2
```
Comment (ravireddy07) [inline Competitive_Programming/C++/Array/K_concatenation_prob.md:109]: remove empty lines

**Review by ravireddy07 — CHANGES_REQUESTED** [review]

File: Competitive_Programming/C++/Array/K_concatenation_prob.md [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+1
+
+3 3
+
+1 3 -2 
+
+**Output 2**
+
+6
+
```
Comment (ravireddy07) [inline Competitive_Programming/C++/Array/K_concatenation_prob.md:126]: remove empty lines(don't leave extra lines)

**Review by ravireddy07 — CHANGES_REQUESTED** [review]

**Comment by vishvarana** [comment]
Bold replaced with heading tags, removed unnecessary spaces except the ones in input and output test cases because they come in a single line if not left an extra line @ravireddy07

**Review by ravireddy07 — APPROVED** [review]

**Pull request merged**